cli/compose/types/types.go

```diff
@@ -125,6 +125,14 @@ type ServiceConfig struct {
 	WorkingDir      string `mapstructure:"working_dir"`
 }
 
+// BuildConfig is a type for build
+// using the same format at libcompose: https://github.com/docker/libcompose/blob/master/yaml/build.go#L12
+type BuildConfig struct {
+	Context    string
+	Dockerfile string
+	Args       map[string]*string
```

Comment:
This looks incomplete. It should have all the fields from here: https://github.com/docker/cli/blob/e6006219abba35fe118d330b1e99b7cf2c9735f1/cli/compose/schema/data/config_schema_v3.4.json#L84-L90

Reply:
@dnephin I'm following @vdemeester 's implementation, but sure, I'll add the other options.